src/main/rx/jetiexbus.c

```diff
@@ -0,0 +1,595 @@
+            jetiExBusFrameLength = jetiExBusFrame[EXBUS_HEADER_MSG_LEN];
+        } else {
+            jetiExBusFrameReset();                  // not a valid frame
+            jetiExBusFrameState = EXBUS_STATE_ZERO;
+            jetiExBusRequestState = EXBUS_STATE_ZERO;
+            return;
+        }
+    }
+
+    // Done?
+    if (jetiExBusFrameLength == jetiExBusFramePosition) {
+        if (crc == 0){
+            if (jetiExBusFrameState == EXBUS_STATE_IN_PROGRESS) jetiExBusFrameState = EXBUS_STATE_RECEIVED;
+            if (jetiExBusRequestState == EXBUS_STATE_IN_PROGRESS) jetiExBusRequestState =  EXBUS_STATE_RECEIVED;
+        }
+        jetiExBusFrameReset();
+    }
+}
+
+
+// Check if it is time to read a frame from the data...
+uint8_t jetiExBusFrameStatus(void)
+{
+    if (jetiExBusFrameState == EXBUS_STATE_RECEIVED) {
+        jetiExBusDecodeFrame(&jetiExBusSerialFrame[0]);
+        jetiExBusFrameState = EXBUS_STATE_ZERO;
+    } else {
+        return SERIAL_RX_FRAME_PENDING;
+    }
+    return SERIAL_RX_FRAME_COMPLETE;
+}
+
+
+static uint16_t jetiExBusReadRawRC(rxRuntimeConfig_t *rxRuntimeConfig, uint8_t chan)
+{
+    if (chan >= rxRuntimeConfig->channelCount)
+        return 0;
+
+    return (jetiExBusChannelData[chan]);
+}
+
+#ifdef TELEMETRY
+/*
+  -----------------------------------------------
+   Jeti Ex Bus Telemetry
+  -----------------------------------------------
+*/
+void createExTelemetrieTextMessage(uint8_t *exMessage, uint8_t messageID, const exBusSensor_t *Sensor)
+{
+    uint8_t labelLength = strlen(Sensor->name);
+    uint8_t unitLength = strlen(Sensor->unit);
+
+    jetiExTelemetry[EXTEL_HEADER_LSN_LB] = messageID & 0xF0;
+
+    exMessage[EXTEL_HEADER_TYPE_LEN] = 8 + labelLength + unitLength;       // 7 header, 1 CRC, label & unit
+    exMessage[EXTEL_HEADER_ID] = messageID & 0x0F;                     // ExMessage ID
+    exMessage[EXTEL_HEADER_DATA] = (labelLength << 3) + unitLength;
+
+    memcpy(&exMessage[EXTEL_HEADER_DATA + 1], Sensor->name, labelLength);
+    memcpy(&exMessage[EXTEL_HEADER_DATA + 1 + labelLength], Sensor->unit, unitLength);
+
+    exMessage[exMessage[EXTEL_HEADER_TYPE_LEN] + 1] = calcCRC8(&exMessage[EXTEL_HEADER_TYPE_LEN], exMessage[EXTEL_HEADER_TYPE_LEN]);
+}
+
+
+uint8_t createExTelemetrieValueMessage(uint8_t *exMessage, uint8_t itemStart)
+{
+    uint8_t lastItem;
+    uint8_t byteCount = 0;
+    uint8_t valueByteCount = 0;
+    uint8_t iCount = 0;
+    uint32_t sensorValue;
+
+
+    if ((itemStart % 16) == 0)
+        itemStart++;
+
+    lastItem = itemStart + 5;
+
+    if (lastItem > JETI_EX_SENSOR_COUNT)
+        lastItem = JETI_EX_SENSOR_COUNT;
+
+    jetiExTelemetry[EXTEL_HEADER_LSN_LB] = itemStart & 0xF0;
+
+    if (itemStart>JETI_EX_SENSOR_COUNT)
+        return 0;
+
+    for (uint8_t item=itemStart; item < lastItem; item++){
+
+        switch(jetiExSensors[item].exDataType) {
```

Comment:
```
uint8_t exDataTypeLen[10] = { 1, 2, 0, 3, 3, 0, 0, 4, 4};
```
This may be written even written using designated initializers to match it directly againt enum:
```
uint8_t exDataTypeLen[10] = {
[EX_TYPE_6b] = 1,
[EX_TYPE_14b] = 2,
...
```